src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+
+  public async getAntoraYmlPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    for (const antoraYmlFile of antoraYmlFiles) {
+      const asciidocFilename = path.basename(pathToAsciidocFile)
+      const modulesPath = path.join(path.dirname(antoraYmlFile.path), 'modules')
+      if (pathToAsciidocFile.startsWith(modulesPath) && pathToAsciidocFile.slice(modulesPath.length).match(/^\/[^/]+\/pages\/.*/) && asciidocFilename !== 'README.adoc') {
+        return antoraYmlFile.path
+      }
+      console.log('Antora file not applicable to current asciidoc document.')
```

Comment:
Let's say we found 5 `antora.yml` and none are applicable, we will log:

```
Antora file not applicable to current asciidoc document.
Antora file not applicable to current asciidoc document.
Antora file not applicable to current asciidoc document.
Antora file not applicable to current asciidoc document.
Antora file not applicable to current asciidoc document.
```

The log message does not provide any context (and it says the same thing over and over). It would be better to log something like:

```
Unable to find an applicable Antora configuration file in [/path/to/antora.yml, /path/to/dir/antora.yml, /path/to/dir/sub/antora.yml] for the AsciiDoc document /path/to/page.adoc" 
```

We could also add a log when we found an applicable antora.yml file:

```
Found an Antora configuration file at /path/to/antora.yml for the AsciiDoc document /path/to/page.adoc
```